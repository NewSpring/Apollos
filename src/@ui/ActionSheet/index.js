export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
