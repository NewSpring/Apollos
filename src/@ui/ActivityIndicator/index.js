import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
