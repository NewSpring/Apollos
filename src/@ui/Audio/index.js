import React, { Component } from 'react';
import { Audio as ExpoAudio } from 'expo';
import PropTypes from 'prop-types';
import { Animated, View } from 'react-native';
import AudioPlay from './AudioPlay';
import AudioPause from './AudioPause';
import AudioSeeker from './AudioSeeker';

export default class Audio extends Component {
  static Play = AudioPlay;
  static Pause = AudioPause;
  static Seeker = AudioSeeker;

  static propTypes = {
    source: PropTypes.string,
    onReady: PropTypes.func,
    onError: PropTypes.func,
    onPlaybackReachedEnd: PropTypes.func,
    onPlay: PropTypes.func,
    onPause: PropTypes.func,
    onStop: PropTypes.func,
    onSeek: PropTypes.func,
    onSeeking: PropTypes.func,
    children: PropTypes.node,
    isPlaying: PropTypes.bool,
    style: PropTypes.any, // eslint-disable-line
  };

  static defaultProps = {
    onReady() {},
    onError(err) {
      // eslint-disable-next-line no-console
      console.error(err.message);
    },
    onPlaybackReachedEnd() {},
    onPlay() {},
    onPause() {},
    onStop() {},
    onSeek() {},
    onSeeking() {},
    children: null,
    isPlaying: false,
  };

  static childContextTypes = {
    play: PropTypes.func,
    stop: PropTypes.func,
    pause: PropTypes.func,
    seek: PropTypes.func,
    progress: PropTypes.object,
    positionMillis: PropTypes.object,
    seekingHandler: PropTypes.func,
  };

  getChildContext = () => ({
    play: this.play,
    stop: this.stop,
    pause: this.pause,
    seek: this.seek,
    progress: this.progressDriver,
    positionMillis: this.positionMillisDriver,
    seekingHandler: this.handleSeeking,
  });

  componentWillMount() {
    ExpoAudio.setAudioModeAsync({
      allowsRecordingIOS: false,
      interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
      playsInSilentModeIOS: true,
      shouldDuckAndroid: true,
      interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX,
    });

    this.loadSource();
  }

  componentDidUpdate({ source, isPlaying }) {
    if (this.props.source !== source) {
      this.loadSource();
    } else if (this.props.isPlaying !== isPlaying) {
      if (this.props.isPlaying) this.play();
      if (!this.props.isPlaying) this.pause();
    }
  }

  componentWillUnmount() {
    if (this.sound) this.sound.unloadAsync();
  }

  onPlaybackStatusUpdate = (soundStatus) => {
    this.positionMillisDriver.setValue(soundStatus.positionMillis);
    this.progressDriver.setValue((soundStatus.positionMillis / this.duration) || 0);

    if (soundStatus.didJustFinish) {
      this.props.onPlaybackReachedEnd(this.sound);
    }
  }

  positionMillisDriver = new Animated.Value(0);
  progressDriver = new Animated.Value(0);
  duration = 0;
  positionListener = undefined;
  previousSoundStatus = undefined;
  isReady = false;
  playState = this.props.isPlaying;

  play = async () => {
    if (this.sound) {
      try {
        await this.sound.playAsync();
        this.props.onPlay();
      } catch (err) {
        this.props.onError(err);
      }
    }
  }

  pause = async () => {
    if (this.sound) {
      try {
        await this.sound.pauseAsync();
        this.props.onPause();
      } catch (err) {
        this.props.onError(err);
      }
    }
  }

  stop = async () => {
    try {
      this.sound.stopAsync();
      this.props.onStop();
    } catch (err) {
      this.props.onError(err);
    }
  }

  seek = (percentageOfSong) => {
    const positionInMillis = this.duration * percentageOfSong;
    this.sound.setPositionAsync(positionInMillis);
    this.props.onSeek(positionInMillis);
  }

  handleSeeking = (percentageOfSong) => {
    const positionInMillis = this.duration * percentageOfSong;
    this.props.onSeeking(positionInMillis);
  }

  loadSource = async () => {
    const {
      source,
      onReady,
      onError,
    } = this.props;

    if (!source) return;
    const uri = source.replace(/^http:\/\/|^\/\//i, 'https://');

    try {
      if (this.sound && this.soundLoaded === this.sound) {
        await this.sound.unloadAsync();
      }

      const sound = this.sound = await new ExpoAudio.Sound(); // eslint-disable-line
      const soundStatus = await this.sound.loadAsync({ uri });
      this.soundLoaded = this.sound;

      if (sound === this.sound) { // since the above calls are promise base, there's a chance that
      // the current sound changes before this sound is loaded (pressing skip button quickly).
      // We don't want to start playing this sound in that scenario :)
        this.duration = soundStatus.durationMillis;
        this.sound.setOnPlaybackStatusUpdate(this.onPlaybackStatusUpdate);
        if (this.props.isPlaying) await this.play();
      }

      onReady();
    } catch (err) {
      onError(err);
    }
  }

  render() {
    return <View style={this.props.style}>{this.props.children}</View>;
  }
}
