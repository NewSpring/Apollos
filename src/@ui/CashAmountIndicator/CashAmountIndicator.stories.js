import React from 'react';

import { storiesOf } from '@storybook/react-native';
import FlexedView from '@ui/FlexedView';

import CashAmountIndicator from './';

storiesOf('CashAmountIndicator', module)
  .add('default', () => (
    <FlexedView>
      <CashAmountIndicator
        amount={2.22}
      />
    </FlexedView>
  ))
  .add('size 2', () => (
    <FlexedView>
      <CashAmountIndicator
        amount={2.22}
        size={2}
      />
    </FlexedView>
  ))
  .add('size 3', () => (
    <FlexedView>
      <CashAmountIndicator
        amount={2.22}
        size={3}
      />
    </FlexedView>
  ))
  .add('size 4', () => (
    <FlexedView>
      <CashAmountIndicator
        amount={2.22}
        size={4}
      />
    </FlexedView>
  ))
  .add('size 5', () => (
    <FlexedView>
      <CashAmountIndicator
        amount={2.22}
        size={5}
      />
    </FlexedView>
  ));
