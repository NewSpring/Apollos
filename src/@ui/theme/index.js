export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
