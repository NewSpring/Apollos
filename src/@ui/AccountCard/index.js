import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
