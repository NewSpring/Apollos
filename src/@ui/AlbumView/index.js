import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { View, Platform, StyleSheet } from 'react-native';
import PaddedView from '@ui/PaddedView';
import ProgressiveImage from '@ui/ProgressiveImage';
import { H5, H7 } from '@ui/typography';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

const Container = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
}), 'AlbumView.Container')(View);

const InfoWrapper = styled({
  flexDirection: 'row', alignItems: 'stretch',
}, 'AlbumView.InfoWrapper')(PaddedView);

const AlbumArt = styled({
  position: 'relative',
  width: '33%',
  aspectRatio: 1,
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '100%',
    },
  }),
}, 'AlbumView.AlbumArt')(ProgressiveImage);

const BlurredImage = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  opacity: theme.alpha.low,
}), 'AlbumView.BlurredImage')(ProgressiveImage);

const TitleWrapper = styled({
  width: '66%',
  justifyContent: 'space-around',
}, 'AlbumView.TitleWrapper')(PaddedView);

const Title = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Title')(H5);

const Artist = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Artist')(H7);

const enhance = compose(
  withThemeMixin({ type: 'dark' }),
  pure,
  setPropTypes({
    title: PropTypes.string,
    artist: PropTypes.string,
    blurredImage: ProgressiveImage.propTypes.source,
    albumImage: ProgressiveImage.propTypes.source,
    isLoading: PropTypes.bool,
  }),
);

const AlbumView = enhance(({
  title, artist, blurredImage, albumImage, isLoading = false,
}) => (
  <Container>
    {(blurredImage) ? <BlurredImage source={blurredImage} /> : null}
    <InfoWrapper>
      <AlbumArt isLoading={isLoading} source={albumImage} />
      <TitleWrapper>
        <Title>{title}</Title>
        <Artist>{artist}</Artist>
      </TitleWrapper>
    </InfoWrapper>
  </Container>
));

export default AlbumView;
