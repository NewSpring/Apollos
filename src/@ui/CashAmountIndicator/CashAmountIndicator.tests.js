import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '@ui/theme';
import CashAmountIndicator from './';

describe('the CashAmountIndicator component', () => {
  it('should render a number', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <CashAmountIndicator
          amount={2}
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a decimal', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <CashAmountIndicator
          amount={2.22}
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render accept different sizes', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <CashAmountIndicator
          amount={2.22}
          size={2}
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});

