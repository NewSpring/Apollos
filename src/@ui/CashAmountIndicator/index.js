import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';

import styled from '@ui/styled';
import { H1, H2, H3, H4, H5, H6, H7 } from '@ui/typography';

const Typography = {
  StyledH1: styled(({ theme }) => ({
    color: theme.colors.text.primary,
  }))(H1),

  StyledH2: styled(({ theme }) => ({
    color: theme.colors.text.primary,
  }))(H2),

  StyledH3: styled(({ theme }) => ({
    color: theme.colors.text.primary,
  }))(H3),

  StyledH4: styled(({ theme }) => ({
    color: theme.colors.text.primary,
  }))(H4),

  StyledH5: styled(({ theme }) => ({
    color: theme.colors.text.primary,
  }))(H5),

  StyledH6: styled(({ theme }) => ({
    color: theme.colors.text.primary,
  }))(H6),

  StyledH7: styled(({ theme }) => ({
    color: theme.colors.text.primary,
    fontFamily: theme.typography.fontFamilySans.bold.default,
  }))(H7),
};

const Row = styled({
  flexDirection: 'row',
  alignItems: 'center',
})(View);

function CashAmountIndicator(props) {
  const amountParts = parseFloat(props.amount)
    .toFixed(2)
    .split('.');
  const SmallType = Typography[`StyledH${props.size + 2}`];
  const LargeType = Typography[`StyledH${props.size}`];
  const styleOverride = props.color ? { color: props.color } : undefined;
  return (
    <Row>
      <SmallType style={styleOverride}>{'$'}</SmallType>
      <LargeType style={styleOverride}>{amountParts[0]}</LargeType>
      <SmallType style={styleOverride}>{`.${amountParts[1]}`}</SmallType>
    </Row>
  );
}

CashAmountIndicator.propTypes = {
  amount: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  size(props, propName, componentName) {
    if (props[propName] > 5) {
      return new Error(`${componentName} ${propName} prop cannot be greater than 5`);
    }
    return null;
  },
  color: PropTypes.string,
};

CashAmountIndicator.defaultProps = {
  amount: 0,
  size: 1,
};

export default CashAmountIndicator;
