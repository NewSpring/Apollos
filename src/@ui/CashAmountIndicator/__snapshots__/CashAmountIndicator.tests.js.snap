// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the CashAmountIndicator component should render a decimal 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 32.4,
        "lineHeight": 41.04,
        "paddingBottom": 2.7,
        "paddingTop": 5.4,
      }
    }
  >
    $
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 52.2,
        "lineHeight": 54.81,
        "paddingTop": 11.7,
      }
    }
  >
    2
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 32.4,
        "lineHeight": 41.04,
        "paddingBottom": 2.7,
        "paddingTop": 5.4,
      }
    }
  >
    .22
  </Text>
</View>
`;

exports[`the CashAmountIndicator component should render a number 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 32.4,
        "lineHeight": 41.04,
        "paddingBottom": 2.7,
        "paddingTop": 5.4,
      }
    }
  >
    $
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 52.2,
        "lineHeight": 54.81,
        "paddingTop": 11.7,
      }
    }
  >
    2
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 32.4,
        "lineHeight": 41.04,
        "paddingBottom": 2.7,
        "paddingTop": 5.4,
      }
    }
  >
    .00
  </Text>
</View>
`;

exports[`the CashAmountIndicator component should render accept different sizes 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 25.2,
        "lineHeight": 31.28,
        "paddingBottom": 3.71,
        "paddingTop": 5.04,
      }
    }
  >
    $
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 42.3,
        "lineHeight": 53.58,
        "paddingBottom": 3.6,
        "paddingTop": 6.3,
      }
    }
  >
    2
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Bold",
        "fontSize": 25.2,
        "lineHeight": 31.28,
        "paddingBottom": 3.71,
        "paddingTop": 5.04,
      }
    }
  >
    .22
  </Text>
</View>
`;
